src/func.cc

```diff
@@ -590,6 +600,10 @@ void ShellFunc(const vector<Value*>& args, Evaluator* ev, string* s) {
     g_command_results.push_back(cr);
   }
   *s += out;
+  Symbol lhs = Intern(".SHELLSTATUS");
+  SimpleVar* var = new SimpleVar(std::to_string(returnCode),
+      VarOrigin::UNDEFINED, ev->CurrentFrame(), ev->loc());
+  lhs.SetGlobalVar(var);
```

Comment:
Why do we need this functionality?

On the surface, this is surprising since we explicitly want to move away from Makefiles, not add new stuff to them.

Reply:
Technically it's not needed, but we're doing some more work in android makefiles as part of the conversion to bazel, where it would be helpful. See https://android-review.googlesource.com/c/platform/build/+/1849978

Reply:
The problem is that once you submit this, you can't tell whether people are going to use this functionality or not because the Makefiles are not under your control.

Asked differently: what benefit does this change bring over the hack in the Makefiles you linked above? Is the extra beauty worth the time spent on this change and the risk of its uses getting out of control?